### PR TITLE
Replace maven-antrun-plugin with jaxb2-maven-plugin for using xjc

### PR DESCRIPTION
**Pull request by vincenzolaudizio**
The module called java-maven-model uses the maven-antrun-plugin for generating the Maven model.
The plugin is configured to execute the xjc tool that needs to be available in the path.

It would it better to remove the "system dependency" in favour of the specific maven plugin jaxb2-maven-plugin.

### CODE REVIEW

File: java-maven-model/pom.xml [hunk 1]
```diff
@@ -73,25 +73,29 @@
...
+          <version>2.2</version>
+          <executions>
+              <execution>
+                  <id>xjc</id>
+                  <goals>
+                      <goal>xjc</goal>
+                  </goals>
+              </execution>
+          </executions>
+          <configuration>
```
Comment (sonartech-externalpr) [inline java-maven-model/pom.xml]: MINOR Make this line start at column 9. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: java-maven-model/pom.xml [hunk 2]
```diff
@@ -73,25 +73,29 @@
...
-                  <arg line="-extension -b ${bindingFile} -p ${javaPackageName} -d ${targetDir} -encoding UTF-8 ${xsdPath}" />
-                </exec>
-              </target>
-            </configuration>
-            <goals>
-              <goal>run</goal>
-            </goals>
-          </execution>
-        </executions>
+          <groupId>org.codehaus.mojo</groupId>
```
Comment (sonartech-externalpr) [inline java-maven-model/pom.xml]: MINOR Make this line start at column 9. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: java-maven-model/pom.xml [hunk 3]
```diff
@@ -73,25 +73,29 @@
...
+          <artifactId>jaxb2-maven-plugin</artifactId>
```
Comment (sonartech-externalpr) [inline java-maven-model/pom.xml]: MINOR Make this line start at column 9. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: java-maven-model/pom.xml [hunk 4]
```diff
@@ -73,25 +73,29 @@
...
+          <version>2.2</version>
```
Comment (sonartech-externalpr) [inline java-maven-model/pom.xml]: MINOR Make this line start at column 9. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: java-maven-model/pom.xml [hunk 5]
```diff
@@ -73,25 +73,29 @@
...
+          <executions>
```
Comment (sonartech-externalpr) [inline java-maven-model/pom.xml]: MINOR Make this line start at column 9. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

**Comment by sonartech-externalpr** [comment]
SonarQube analysis reported 5 issues
* MINOR 5 minor

Watch the comments in this conversation to review them.

**Comment by m-g-sonar** [comment]
Hey @vincenzolaudizio,

Thanks a lot for your contribution. Indeed, relying on {{jaxb2-maven-plugin}} seems to be a nicer approach that executing xjc through the ant plugin. 

I just opened PR #1046 with your very same commit, in order to trigger our internal QA and run the ITs. If everything is OK, I'll merge your PR with our master, but not before you fixed the indentation issues which have been raised. Can you please have a look at it? For you information, we use 2-spaces tabs.

Cheers,
Michael

**Comment by m-g-sonar** [comment]
Merged. Again, thanks a lot for your work! :)

**Pull request merged**